packages/obonode/obojobo-chunks-materia/server/route-helpers.js

```diff
@@ -40,16 +39,40 @@ const ltiToolConsumer = baseUrl => ({
 	tool_consumer_instance_url: baseUrl
 })
 
-const widgetLaunchParams = (document, materiaChunkId, user, visitId, isPreview, baseUrl) => {
+const widgetLaunchParams = (document, visit, user, materiaChunkId, baseUrl) => {
 	const params = {
-		lis_outcome_service_url: `${baseUrl}/materia-lti-score-passback`,
 		lti_message_type: 'basic-lti-launch-request',
-		lis_result_sourcedid: createLisResultSourcedId(visitId, materiaChunkId),
 		lti_version: 'LTI-1p0',
-		resource_link_id: materiaChunkId
+		// materia will send scores here
+		lis_outcome_service_url: `${baseUrl}/materia-lti-score-passback`,
+		// represents the container for the user's score this widget can set/update
+		lis_result_sourcedid: `${visit.id}__${materiaChunkId}`,
+		// unique placement of this widget (take into account the module's unique placement in a course + the widget's place in the module)
+		// we are intentionally not including user id (not part of the placement) or draft_content_id (to allow updates to a module)
+		resource_link_id: `${visit.resource_link_id}__${visit.draft_id}__${materiaChunkId}`,
```

Comment:
I was thinking we were gonna use something like

```suggestion
		resource_link_id: `${visit.id}__${materiaOboNodeId}`,
```

As in same as the `lis_result_sourcedid`, since this _would_ account for draft versions and would "reset" the Materia widget on each visit, more mimicking the way practice questions work. (You could easily argue in either direction, but I'm thinking that visit id is where we landed).

Reply:
agreed, changed to this as visit id represents user + launch + draft + draft version.

resource_link_id shouldn't include the user's session by its definition.  

Which I think - means that materia should probably window it's scores/attempts around lis_result_sourcedid